front/lib/resources/group_resource.ts

```diff
@@ -114,6 +114,39 @@ export class GroupResource extends BaseResource<GroupModel> {
     return defaultGroup;
   }
 
+  /**
+   * Checks if an agent editors group exists for the given agent and creates one if it doesn't.
+   * This prevents duplicate group creation when drafting the same agent multiple times.
+   */
+  static async findOrCreateAgentEditorsGroup(
+    auth: Authenticator,
+    agent: AgentConfiguration,
+    { transaction }: { transaction?: Transaction } = {}
+  ) {
+    const workspace = auth.getNonNullableWorkspace();
+
+    if (agent.workspaceId !== workspace.id) {
+      throw new DustError(
+        "internal_error",
+        "Unexpected: agent and workspace mismatch"
+      );
+    }
+
+    const existingAssociation = await GroupModel.findOne({
+      where: {
+        name: `Group for Agent ${agent.name}`,
```

Comment:
Initially wanted to go for a:
```
  const existingAssociation = await GroupAgentModel.findOne({
    where: {
      agentConfigurationId: agent.id,
      workspaceId: agent.workspaceId
    },
    transaction
  });
```
but issue is that the agentConfigurationId might be different from the one that was initially created. Ideally we would go have store the agent `sId`

Reply:
I have little context on this new model but this indeed looks fishy, shouldn't we rework the model to avoid that @philipperolet ? 🙏🏻